testing/test_terminal.py

```diff
@@ -97,6 +97,40 @@ def test_func():
             ["    def test_func():", ">       assert 0", "E       assert 0"]
         )
 
+    @pytest.mark.parametrize(
+        "write_pytest_ini,specify_path",
+        [
+            (False, False),
+            (False, True),
+            (True, False),
+            pytest.param(True, True, marks=pytest.mark.xfail),
+        ],
+    )
+    def test_parametrize_relative_path(self, testdir, write_pytest_ini, specify_path):
```

Comment:
I find this test incredibly difficult to read, please [avoid logic in tests](https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html) -- they should be clear and readable

In this case there's a lot of things that make this particularly hard to comprehend:
- [boolean trap](https://ariya.io/2011/08/hall-of-api-shame-boolean-trap) `parametrize`
- branching logic in tests
- conditional xfail
- splat of ternary list

I'd suggest writing this as separate tests (I'm not even sure all four combinations are helpful here, you can probably just use the failing combination since all the other variants are already passing)

Reply:
FTR I've wrote many of tests using logic like this one, and nowadays I tend to agree with @asottile that it should be avoided in general, using free functions to reuse logic instead.

Reply:
I used this layout to document the specific case that causes the issue, and what cases that do not. I intend to remove the other parameterize entries once it's fixed

Reply:
Oh OK sounds good. 👍 

Do we intend to merge this as is then? If so I think we should add a comment stating the above (removing the parametrization cases that pass).

Reply:
sure, can you `` ```suggestion `` what you think the comment should be?

I'd imagine we'd change this test a lot when the root cause is determined

Reply:
Sure